iotdb/src/main/java/org/apache/iotdb/db/conf/adapter/IoTDBConfigDynamicAdapter.java

```diff
@@ -0,0 +1,226 @@
+
+  // static parameter section
+
+  private static final double FLUSH_THRESHOLD = 0.2;
+
+  /**
+   * Maximum amount of memory allocated for write process.
+   */
+  private static final long ALLOCATE_MEMORY_FOR_WRITE = CONFIG.getAllocateMemoryForWrite();
+
+  /**
+   * Metadata size of per timeseries, the default value is 2KB.
+   */
+  private static final long TIMESERIES_METADATA_SIZE_IN_BYTE = 2L * 1024;
+
+  /**
+   * Metadata size of per chunk, the default value is 1.5 KB.
+   */
+  private static final long CHUNK_METADATA_SIZE_IN_BYTE = 1536L;
+
+  /**
+   * Average queue length in memtable pool
+   */
+  static final int MEM_TABLE_AVERAGE_QUEUE_LEN = 5;
+
+  // static memory section
+
+  /**
+   * Static memory, includes all timeseries metadata, which equals to TIMESERIES_METADATA_SIZE_IN_BYTE *
+   * totalTimeseriesNum, the unit is byte
+   */
+  private long staticMemory;
+
+  private int totalTimeseries;
+
+  // MemTable section
+
+  private int maxMemTableNum = MEM_TABLE_AVERAGE_QUEUE_LEN;
+
+  private int currentMemTableSize;
+
+  // Adapter section
+
+  private boolean initialized = false;
+
+  @Override
+  public synchronized boolean tryToAdaptParameters() {
+    boolean shouldAdjust = true;
+    int memtableSizeInByte = calcMemTableSize();
+    int memTableSizeFloorThreshold = getMemTableSizeFloorThreshold();
+    boolean shouldClose = false;
+    long tsFileSize = CONFIG.getTsFileSizeThreshold();
+    if (memtableSizeInByte < memTableSizeFloorThreshold) {
+      shouldClose = true;
+      tsFileSize = calcTsFileSize(memTableSizeFloorThreshold);
+      memtableSizeInByte = (int) tsFileSize;
+      if (tsFileSize < memTableSizeFloorThreshold) {
+        shouldAdjust = false;
+      }
+    } else if (memtableSizeInByte > tsFileSize) {
+      memtableSizeInByte = (int) tsFileSize;
+    }
+
+    if (shouldAdjust) {
+      CONFIG.setMaxMemtableNumber(maxMemTableNum);
+      CONFIG.setTsFileSizeThreshold(tsFileSize);
+      CONFIG.setMemtableSizeThreshold(memtableSizeInByte);
+      if(initialized) {
+        if (shouldClose) {
+          StorageEngine.getInstance().asyncFlushAndSealAllFiles();
+        } else if (memtableSizeInByte < currentMemTableSize
+            && currentMemTableSize - memtableSizeInByte > currentMemTableSize * FLUSH_THRESHOLD) {
+          StorageEngine.getInstance().asyncFlushAllProcessor();
+        }
+      }
+      currentMemTableSize = memtableSizeInByte;
+    }
+    if (!initialized) {
+      CONFIG.setMaxMemtableNumber(maxMemTableNum);
+      return true;
+    }
+    return shouldAdjust;
+  }
+
+  /**
+   * Calculate appropriate MemTable size
+   *
+   * @return MemTable size. If the value is -1, there is no valid solution.
+   */
+  private int calcMemTableSize() {
+    double ratio = CompressionRatio.getInstance().getRatio();
+    // when unit is byte, it's likely to cause Long type overflow. so use the unit KB.
+    double a = (long) (ratio * maxMemTableNum);
```

Comment:
do not use a, b.... use some words that are meaningful

Reply:
They are intermediate variables used in solving formulas for quadratic equation of one variable, which have no meaning.